torch/lib/c10d/reducer.cpp

```diff
@@ -1036,6 +1036,22 @@ void Reducer::prepare_for_backward(
           unused_parameters_.end(), indices.begin(), indices.end());
     }
   }
+
+  // Warn user about unnecessary perf hit if all parameters were used.
+  if (unused_parameters_.empty() && !foundUnusedParametersOnce_) {
```

Comment:
Do we need the additional variable `foundUnusedParametersOnce_`? Shouldn't `TORCH_WARN_ONCE` take care of this?

Reply:
Yes it should, thanks for the catch